src/screens/Movie/Movie.tsx

```diff
@@ -49,23 +50,31 @@ const Movie = ({ match, location }: RouteComponentProps<MovieRouteParams>): JSX.
 
   // Media
   const { isLoading, error, data: item } = useMedia(id);
-  const itemRequiresSubscription = item?.requiresSubscription !== 'false';
   useBlurImageUpdater(item);
   const { data: trailerItem } = useMedia(item?.trailerId || '');
   const { data: playlist } = useRecommendedPlaylist(recommendationsPlaylist || '', item);
 
-  const isFavorited = useFavoritesStore((state) => !!item && state.hasItem(item));
+  // AccessModel & entitlement
+  const mediaOffer = useMemo(() => (item?.productIds && filterCleengMediaOffers(item.productIds)) || undefined, [item]);
+  useEntitlement(mediaOffer);
 
+  const isFavorited = useFavoritesStore((state) => !!item && state.hasItem(item));
   const watchHistoryItem = useWatchHistoryStore((state) => item && state.getItem(item));
   const progress = watchHistoryItem?.progress;
 
   // General state
   const [hasShared, setHasShared] = useState<boolean>(false);
   const [playTrailer, setPlayTrailer] = useState<boolean>(false);
 
-  // User
+  // User, accessModel, entitlement
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
-  const allowedToWatch = isAllowedToWatch(accessModel, !!user, itemRequiresSubscription, !!subscription);
+  const isItemFree = item?.requiresSubscription === 'false' || !!item?.free;
```

Comment:
Personally, I think I would prefer to put this sort of basic data transformation at a lower level (in this case in the hook) so that components don't have to worry about data format, but can just directly use what they need.